presto-main/src/test/java/io/prestosql/sql/analyzer/TestAnalyzer.java

```diff
@@ -253,10 +256,42 @@ public void testHavingReferencesOutputAlias()
     }
 
     @Test
-    public void testWildcardWithInvalidPrefix()
+    public void testSelectAllColumns()
     {
+        // wildcard without FROM
+        assertFails("SELECT *")
+                .hasErrorCode(COLUMN_NOT_FOUND);
+
+        // wildcard with invalid prefix
         assertFails("SELECT foo.* FROM t1")
                 .hasErrorCode(TABLE_NOT_FOUND);
+
+        assertFails("SELECT a.b.c.d.* FROM t1")
+                .hasErrorCode(TABLE_NOT_FOUND);
+
+        // name aliases are not allowed when selecting all columns from a table
+        assertFails("SELECT t.* AS (a, b, c) FROM (VALUES (1, 2, 3), (10, 20, 30), (100, 200, 300)) t")
+                .hasErrorCode(NOT_SUPPORTED);
+
+        // aliases mismatch
+        assertFails("SELECT (1, 2).* AS (a) FROM t1")
+                .hasErrorCode(MISMATCHED_COLUMN_ALIASES);
+
+        // duplicate alias
+        assertFails("SELECT (1, 2).* AS (a, a) FROM t1")
+                .hasErrorCode(DUPLICATE_COLUMN_NAME);
+
+        // wildcard with no RowType expression
+        assertFails("SELECT non_row.* FROM (VALUES ('true', 1)) t(non_row, b)")
+                .hasErrorCode(TABLE_NOT_FOUND);
+
+        // wildcard with no RowType expression nested in a row
+        assertFails("SELECT t.row.non_row.* FROM (VALUES (CAST(ROW('true') AS ROW(non_row boolean)), 1)) t(row, b)")
+                .hasErrorCode(INVALID_ALL_COLUMNS_TARGET);
```

Comment:
This should probably be a TYPE_MISMATCH error. Basically, it expected a ROW type but found something else.